### PR TITLE
Support unified `android_library` and `android_binary` macros from bazel-common

### PR DESCRIPTION
**Pull request by arunkumar9t2**
Support unified `android_library` and `android_binary` macros from bazel-common

Context https://github.com/grab/grab-bazel-common/pull/87

As result of bazel-common's macro handling more specific implementation cases, the complexity of grazel managing multiple intermediate targets for kotlin, res value and build config has been reduced/removed.

This PR also introduces proper type for representing bazel targets and intermediate extraction data classes `AndroidData`

### CODE REVIEW

File: grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/migrate/builder/KtAndroidLibTargetBuilder.kt [hunk 1]
```diff
@@ -94,6 +92,6 @@ internal class KtAndroidLibTargetBuilder @Inject constructor(
     }
 
     override fun canHandle(project: Project): Boolean = with(project) {
-        isAndroid && isKotlin && !isAndroidApplication
+        isAndroid && isKotlin && !isAndroidApplication && false
```
Comment (arunkumar9t2) [inline grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/migrate/builder/KtAndroidLibTargetBuilder.kt:95]: The whole KtAndroidBuilder will be removed in another PR

**Review by mohamadk — APPROVED** [review]
